components/camel-xslt-saxon/src/main/java/org/apache/camel/component/xslt/saxon/XsltSaxonEndpoint.java

```diff
@@ -209,8 +209,7 @@ protected XsltSaxonBuilder createXsltBuilder() throws Exception {
             }
         }
 
-        if (factory instanceof TransformerFactoryImpl) {
-            TransformerFactoryImpl tf = (TransformerFactoryImpl) factory;
+        if (factory instanceof TransformerFactoryImpl tf) {
```

Comment:
This code style is only java 17 and as such this feature cannot be backported to 3.x. 

This PR is not related to this change, so its better to keep code as before, as it makes backporting bugs and other PRs easier from v4 to v3

Reply:
I'll revert this.